torch/quantization/convert_modules.py

```diff
@@ -0,0 +1,183 @@
+    module.register_forward_pre_hook(_quant_forward_pre_hook)
+    module.register_forward_hook(_dequant_forward_hook)
+
+def add_quant_dequant_module(module, qconfig):
+    module.add_module('quant', QuantStub(qconfig))
+    module.add_module('dequant', DeQuantStub())
+    add_quant_dequant_hooks(module)
+
+def add_quant_dequant(module, qconfig_dict):
+    r"""Add QuantStub and DeQuantStub module and add quant dequant calls using
+    forward hooks
+    """
+    mod_key_set = set()
+    for name, _ in module.named_modules():
+        dict_key = get_config_key(name, qconfig_dict)
+        if dict_key is not None:
+            mod_key_set.add(dict_key)
+
+    for name in mod_key_set:
+        add_quant_dequant_module(get_module(module, name), qconfig_dict[name])
+    return module
+
+def prepare(module, qconfig_dict):
+    r"""Prepares the module for calibration according to qconfig_dict.
+
+    The function adds observer and quant dequant stub modules to the module.
+    It also adds the forward hooks to call quant and dequant methods.
+
+    Args:
+        mod: input module
+        qconfig_dict: dictionary that maps from name of submodule to quantization
+                      configuration
+    """
+    add_observer(module, qconfig_dict)
+    add_quant_dequant(module, qconfig_dict)
+
+class QuantStub(nn.Module):
+    r"""Quantize stub module which replaces the quantized module
+    in `swap_module` function.
+    """
+    def __init__(self, qconfig):
+        super(QuantStub, self).__init__()
+        self.add_module('observer', qconfig.activation())
+
+    def forward(self, x):
+        self.observer(x)
+        return x
+
+class DeQuantStub(nn.Module):
+    r"""Dequantizes stub module, which replaces the actual DeQuantize module
+    in `swap_module` function.
+    """
+    def __init__(self):
+        super(DeQuantStub, self).__init__()
+
+    def forward(self, x):
+        return x
+
+def quantize(module, qconfig_dict, eval_fn, eval_args):
+    r"""Converts a float module to quantized module.
+
+    First it will prepare the module for calibration or training, then it calls
+    `eval_fn` which will run the calibration step or training step,
+    after that we will call `convert` which will convert the module to a
+    quantized module.
+    """
+    prepare(module, qconfig_dict)
+    eval_fn(module, *eval_args)
+    convert_to_quantized(module)
+    return module
+
+def convert_to_quantized(module):
+    r"""Converts the float module with qparams to a quantized module.
+    """
+    module_swapped = swap_module(module)
+    if len(list(module.named_children())) == 0:
+        return module_swapped
+
+    reassign = {}
+    for name, mod in module.named_children():
+        new_mod = convert_to_quantized(mod)
+        if new_mod is not mod:
+            reassign[name] = new_mod
+
+    for name, mod in reassign.items():
+        setattr(module_swapped, name, mod)
+
+    return module_swapped
+
+DEFAULT_MODULE_MAPPING = {
```

Comment:
Related to my previous comment: These need not be arguments to the quantize function, can instead be dictionaries like here.